sdk/api/sdk.api

```diff
@@ -0,0 +1,265 @@
+	public static fun values ()[Lio/customer/sdk/hooks/HookModule;
+}
+
+public abstract interface class io/customer/sdk/hooks/HooksManager {
+	public abstract fun add (Lio/customer/sdk/hooks/HookModule;Lio/customer/sdk/hooks/ModuleHookProvider;)V
+	public abstract fun onHookUpdate (Lio/customer/sdk/hooks/ModuleHook;)V
+}
+
+public abstract class io/customer/sdk/hooks/ModuleHook {
+}
+
+public final class io/customer/sdk/hooks/ModuleHook$BeforeProfileStoppedBeingIdentified : io/customer/sdk/hooks/ModuleHook {
+	public fun <init> (Ljava/lang/String;)V
+	public final fun getIdentifier ()Ljava/lang/String;
+}
+
+public final class io/customer/sdk/hooks/ModuleHook$ProfileIdentifiedHook : io/customer/sdk/hooks/ModuleHook {
+	public fun <init> (Ljava/lang/String;)V
+	public final fun getIdentifier ()Ljava/lang/String;
+}
+
+public final class io/customer/sdk/hooks/ModuleHook$ScreenTrackedHook : io/customer/sdk/hooks/ModuleHook {
+	public fun <init> (Ljava/lang/String;)V
+	public final fun getScreen ()Ljava/lang/String;
+}
+
+public abstract class io/customer/sdk/hooks/ModuleHookProvider {
+	public fun <init> ()V
+	public fun beforeProfileStoppedBeingIdentified (Lio/customer/sdk/hooks/ModuleHook$BeforeProfileStoppedBeingIdentified;)V
+	public fun profileIdentifiedHook (Lio/customer/sdk/hooks/ModuleHook$ProfileIdentifiedHook;)V
+	public fun screenTrackedHook (Lio/customer/sdk/hooks/ModuleHook$ScreenTrackedHook;)V
+}
+
+public abstract interface class io/customer/sdk/repository/DeviceRepository {
+	public abstract fun addCustomDeviceAttributes (Ljava/util/Map;)V
+	public abstract fun deleteDeviceToken ()V
+	public abstract fun registerDeviceToken (Ljava/lang/String;Ljava/util/Map;)V
+}
+
+public abstract interface class io/customer/sdk/repository/PreferenceRepository {
+	public abstract fun clearAll ()V
+	public abstract fun getDeviceToken ()Ljava/lang/String;
+	public abstract fun getHttpRequestsPauseEnds ()Ljava/util/Date;
+	public abstract fun getIdentifier ()Ljava/lang/String;
+	public abstract fun removeIdentifier (Ljava/lang/String;)V
+	public abstract fun saveDeviceToken (Ljava/lang/String;)V
+	public abstract fun saveIdentifier (Ljava/lang/String;)V
+	public abstract fun setHttpRequestsPauseEnds (Ljava/util/Date;)V
+}
+
+public abstract interface class io/customer/sdk/repository/ProfileRepository {
+	public abstract fun addCustomProfileAttributes (Ljava/util/Map;)V
+	public abstract fun clearIdentify ()V
+	public abstract fun identify (Ljava/lang/String;Ljava/util/Map;)V
+}
+
+public abstract interface class io/customer/sdk/repository/TrackRepository {
+	public abstract fun screen (Ljava/lang/String;Ljava/util/Map;)V
+	public abstract fun track (Ljava/lang/String;Ljava/util/Map;)V
+	public abstract fun trackInAppMetric (Ljava/lang/String;Lio/customer/sdk/data/request/MetricEvent;)V
+	public abstract fun trackMetric (Ljava/lang/String;Lio/customer/sdk/data/request/MetricEvent;Ljava/lang/String;)V
+}
```

Comment:
Do repositories need to be public?

Reply:
Yes, as they can get accessed and used by other modules e.g `ModuleMessagingInApp` etc